app/lib/travis/TravisApiClient.scala

```diff
@@ -92,41 +104,27 @@ class TravisApiClient(githubToken: String) extends LazyLogging {
       )
     )
 
-    for {
-      authToken <- authTokenSupplier.get()
-      response <- postJson[JsValue, JsValue](
-        s"/repo/${URLEncoder.encode(repoId, "UTF-8")}/requests",
-        bodyJson,
-        Seq(
-          "Authorization" -> s"token $authToken",
-          "Travis-API-Version" -> "3"
-        )
-      )
-    } yield response
+    postJson[JsValue, JsValue](s"/repo/${URLEncoder.encode(repoId, "UTF-8")}/requests", bodyJson)
   } andThen { case respTry => logger.info(s"requestBuild on $repoId response=$respTry") }
 
+}
 
-  val authTokenSupplier = new AuthTokenSupplier[String](auth(githubToken).map(_.get.access_token))
-
-  class AuthTokenSupplier[T](generateToken: => Future[T]) {
-
-    private val agent: Agent[Future[T]] = Agent(generateToken)
+sealed trait TravisCIOffering {
+  val name: String
+  val baseApiEndpoint: Uri
+}
 
-    /**
-     * If no refresh is currently running, start a new one, return that future
-     * If a refresh is currently running, return the existing future
-     */
-    def refresh(): Future[T] = for {
-      refreshAlteringFuture <- agent.alter { currentRefresh => if (currentRefresh.isCompleted) generateToken else currentRefresh }
-      refreshFuture <- refreshAlteringFuture
-    } yield refreshFuture
+object TravisCIOffering {
```

Comment:
The `OpenSource` & `Commercial` objects below are the _"stateless objects that just encapsulate the API url"_ that I was talking about in https://github.com/guardian/prout/pull/57#discussion_r156678804 - they don't have state, they just do the sole job of expressing the difference between `travis-ci.org` (free) & `travis-ci.com` (commercial) offerings.